### ISSUE TITLE
Failed to Compile 

### ISSUE THREAD

**Issue by JonTron1987** [vercel/next-learn#72]
I get an error when running npm dev run. I have posted the image of what it is saying instead of typing it out. I follow the instructions per the NEXT.js website. What do I need to do to fix this error?

![image](https://user-images.githubusercontent.com/12778501/161401503-0342838e-f2a5-48f0-8cd8-3fac8cc86ecb.png)

**Comment by mroandersons** [comment]
Same here. Did you find solution?

**Comment by JonTron1987** [comment]
no...currently searching...but so far no luck

    On Saturday, April 9, 2022, 11:02:11 AM EDT, mroandersons ***@***.***> wrote:  
 
 


Same here. Did you find solution?

**Comment by Grimnir09** [comment]
Ditto.

**Comment by JonTron1987** [comment]
Hm..let me try that. Will see what happens and follow up.

**Comment by JonTron1987** [comment]
yeah, for some reason I am not getting that issue with a default template build; it was just the one from the Next.js site example that had the error. Quite interesting.

**Comment by BradleyCulley** [comment]
I just wrote a fix for that and created a pull request with the fix: https://github.com/vercel/next-learn/pull/75

**Comment by BradleyCulley** [comment]
> yeah, for some reason I am not getting that issue with a default template build; it was just the one from the Next.js site example that had the error. Quite interesting.

I just wrote a fix for that and created a pull request with the fix: https://github.com/vercel/next-learn/pull/75

**Comment by huozhi** [comment]
@BradleyCulley the styled-jsx is supported out of box in next.js, adding duplicated styled-jsx will lead to more issues.

@JonTron1987 Could you share more about how to create the starter? The commands you run, and the package manager version you're using.

And could you also share the result of running `npm ls styled-jsx` under your project? Thanks

**Comment by BradleyCulley** [comment]
> @BradleyCulley the styled-jsx is supported out of box in next.js, adding duplicated styled-jsx will lead to more issues.
> 
> @JonTron1987 Could you share more about how to create the starter? The commands you run, and the package manager version you're using.
> 
> And could you also share the result of running `npm ls styled-jsx` under your project? Thanks

@huozhi Thank you for your reply. I made a short narrated video demonstrating the reproducible sequence for this issue, doing that step of the tutorial freshly, in an empty directory. I thought that would be an efficient way to demonstrate it. This is the video: https://vimeo.com/704367470.

**Comment by BradleyCulley** [comment]
@huozhi Thank you for your reply. I ran through those items in another narrated video, here: https://vimeo.com/704698185. Perhaps I'm missing something in your recommendations, or my node/npm version is incompatible. For reference, I've been doing React professionally since 2016, and frontend software engineering for 13 years: https://www.linkedin.com/in/brad-culley-93651992/. So I'm new to Next, but not React/npm/etc. I may be missing something obvious though.

**Comment by huozhi** [comment]
@BradleyCulley Thaks for the feedback, Sorry for the confusion for not explaining the reason clearly of asking for those system info. 
I asked for those information because we had some problem with previous npm versions like (7.20.x - 8.1.0) that sometimes installed styled-jsx incorrectly. 
And I asked for version of CRN because npx might cache the previous verisons that didn't use the latest one which might have bugs that are already been fixed. 
The command `npm ls styled-jsx` could be helpful for checking if you might installed duplicated styled-jsx.


The suspectious one is the npm version, would you mind upgrading npm to any version above 8.1.0 to see if it still existed?

**Comment by huozhi** [comment]
I can stablely repro locally with `npm@8.1.0` now that styled-jsx isn't installed under `<project dir>/node_modules` but `<project dir>/next/node_modules` which breaks the node module resolving with styled-jsx.

After bumping to npm versions above 8.1.0 the issue is gone for me

**Comment by BradleyCulley** [comment]
@huozhi  That works! Thank you Jiachi. This is a short video showing it working: https://vimeo.com/704712718. I installed latest Node LTS and got npm 8.5.5 with it.

Would it be worth adding to the docs that one's npm version must be greater than 8.1.0?

**Comment by nipunn1313** [comment]
I too ran into this issue. Bumping to NPM 8 solved.
I was previously on
```
➜ npm --version
7.24.2
```

The failure mode this hits when you're accidentally on an old version of NPM is pretty hard to figure out - especially since this is the tutorial brand new folks are pointed to https://nextjs.org/learn/basics/create-nextjs-app. It's probably unreasonable to expect brand new folks to debug NPM versions.

Fixed by bumping to
```
➜ npm --version
8.8.0
``

**Comment by BradleyCulley** [comment]
@nipunn1313 Agreed it's a bit of a pitfall of the tutorial. @huozhi would it be worth adding to the tutorial that one's npm version must be greater than 8.1.0?

**Comment by huozhi** [comment]
We improved the support of styled-jsx in next.js for pnpm/yarnPnP in `next@12.1.6-canary.16`, ideally the `create-next-app@canary` could solve the issue as well, also works with npm 8.1.0.

Could you also try the command with canary tag to see if the problem is gone? If it works then we could close the issue and pr.

```
npx create-next-app@canary nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"
```

It will also go out with the next stable version so the we can leave the doc as it rn

**Comment by BradleyCulley** [comment]
> We improved the support of styled-jsx in next.js for pnpm/yarnPnP in `next@12.1.6-canary.16`, ideally the `create-next-app@canary` could solve the issue as well, also works with npm 8.1.0.
> 
> Could you also try the command with canary tag to see if the problem is gone? If it works then we could close the issue and pr.
> 
> ```
> npx create-next-app@canary nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"
> ```
> 
> It will also go out with the next stable version so the we can leave the doc as it rn

@huozhi Hi Jiachi - I just ran the command with the canary version, and get the same error. Here is a video demonstrating that: https://vimeo.com/705192965

**Comment by huozhi** [comment]
hey @BradleyCulley thanks for the repro I was missing 1 step, which is:

```
# edit package.json to "next": "latest" to "next": "canary"

# then run install
npm install
```

The CRN will always install the latest version of next. If you switch to canary version it works. Make sure you clean the package-lock.json and node_modules before the re-installation!

**Comment by BradleyCulley** [comment]
> hey @BradleyCulley thanks for the repro, and sorry I was missing 1 step, which is:
> 
> ```
> # edit package.json to "next": "latest" to "next": "canary"
> 
> # then run install
> npm install
> ```
> 
> Added to previous comment
> 
> The CRN will always install the latest version of next. If you switch to canary version it works. Make sure you clean the package-lock.json and node_modules before the re-installation!

Hi @huozhi - that works great. That sounds like a good resolution of the issue: it'll be fixed once the current canary build gets promoted to latest.

**Comment by huozhi** [comment]
The module resolving improvement is landed in next 12.0.6 or upgrading npm (> 8.1.0) can also fix this issue
